### PR TITLE
Add 99717-49 Local Code to 76030-6 LOINC mapping

### PR DESCRIPTION
**Pull request by jbiskie**
# Adds 99717-49 Local Code to 76030-6 LOINC mapping

This PR:
- Adds a new local to LOINC mapping
- Refactors the hardcoded mapping in the transform class into a configurable arguments in `transformation_definitions.json`
- Adds additional unit tests to cover cases where the codingMap json argument is not structured as expected

## Issue

#1452 

This PR covers engineering tasks 1 and 2. Task 3 (Add RS E2E tests for the mapping transformation) will be addressed in another PR.

## Checklist

- [x] I have added tests to cover my changes
- [x] I have added logging where useful (with appropriate log level)
- [x] I have added JavaDocs where required
- [x] I have updated the documentation accordingly

**Note**: You may remove items that are not applicable

### CODE REVIEW

**Comment by jbiskie** [comment]
> ## PR Code Suggestions ✨
> 
> Explore these optional code suggestions:
> Category	Suggestion                                                                                                                                    	Score
> **Possible issue** 	
> Prevent potential overwriting of existing keys in the map
> 
> 	7

Non-issue. IntelliJ flags duplicates.

**Comment by halprin** [comment]
We should also update our RS E2E tests to test this.

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/MapLocalObservationCodes.java [hunk 1]
```diff
@@ -23,52 +23,68 @@
...
+        if (field == null || field.isBlank()) {
+            throw new IllegalArgumentException("missing or empty required field " + fieldName);
         }
+        return field;
     }
 
-    private void logUnmappedLocalCode(Bundle bundle, Coding coding) {
-        var msh41Identifier = HapiHelper.getMSH4_1Identifier(bundle);
-        var msh41Value = msh41Identifier != null ? msh41Identifier.getValue() : null;
+    private void logUnmappedLocalCode(Coding coding, String msh41Identifier, String messageId) {
```
Comment (basiliskus) [inline etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/MapLocalObservationCodes.java:81]: Looks like this is only used once, it has a single line and the content is pretty straightforward. Does it need to extracted into its own function?

**Comment by jbiskie** [comment]
Manual RS end to end testing was successful. Used file 020 and added an extra OBX for 99717-49:

`OBX|12|TX|^^^99717-49^New local code added^L|1|Negative|||N|||F|||20240711034913`

transformed:

`OBX|113|TX|76030-6^IDS gene full mutation analysis in Blood or Tissue by Sequencing^LN^99717-49^New local code added^L|1|Negative|||N|||F|||20240711034913`

**Review by basiliskus — COMMENTED** [review]
Seems like there are some extracted functions that have a single line of code and are used only once. Are they needed? There's a tradeoff when refactoring to extract code to functions and adding too many can create unnecessary overhead. If they are added to help understand what the code does, you could consider adding a comment instead

**Review by luis-pabon-tf — APPROVED** [review]

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/MapLocalObservationCodes.java (hunk 1 above)
Comment (jorg3lopez) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/MapLocalObservationCodes.java:81]: Good suggestion. I'll inline the extracting method and keep the logging one as is to keep the logging logic in one area.

Comment (basiliskus) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/MapLocalObservationCodes.java:79]: What I meant is that we probably don't need `logUnmappedLocalCode` as it's adding more abstraction without the benefits. You could directly call `logger.logWarning` without putting it into its own function

Comment (jorg3lopez) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/MapLocalObservationCodes.java]: Made the logging inline. Thanks for the suggestion, I think it makes sense.

**Review by basiliskus — APPROVED** [review]

**Pull request merged**